README.md

```diff
@@ -1,2 +1,91 @@
-# rspec-sequencing
-Define sequenced actions that simulate real-world scenarios, e.g write_file then 2 seconds later rename it
+# Rspec::Sequencing
+
+Define sequenced actions that simulate real-world scenarios, e.g. write_file then 2 seconds later rename it or execute a sequence of actions on a http server
+
+## Installation
+
+Add this line to your application's Gemfile:
+
+```ruby
+gem 'rspec-sequencing'
+```
+
+And then execute:
+
+    $ bundle
+
+Or install it yourself as:
+
+    $ gem install rspec-sequencing
+
+## Usage
+
+Generally one would use this when you need to have a series of actions occur in way that reflects the real world in terms of being asynchronous, serial, and spread out over time especially when the library code you are testing has difficult to mock dependencies like sockets, files, servers or apis, i.e. integration testing.
+
+Add `require "rspec_sequencing"` in your spec_helper.rb or equivalent.
+
+In your example group you can define a sequence in a before block or a let block.  If you use a let block, then you should call the activate method on the sequence to get RSpec to create it. Call activate in a before block or in the example.
+
+e.g. from the ruby-filewatch gem watch_spec.rb
+```ruby
+  context "when watching a directory with files" do
+    let(:actions) do
+      RSpec::Sequencing
+        .run("create file") do
+          File.open(file_path, "wb") { |file|  file.write("line1\nline2\n") }
+        end
+        .then_after(0.25, "start watching when directory has files") do
+          subject.watch(File.join(directory, "*.log"))
+        end
+        .then_after(0.55, "quit after a short time") do
+          subject.quit # <- unblocks the RSpec thread
+        end
+    end
+
+    it "yields create_initial and one modify file events" do
+      actions.activate
+      subscribe_proc.call # <- blocks the RSpec thread
+      expect(results).to eq([[:create_initial, file_path], [:modify, file_path]])
```

Comment:
Its all so very dependent on the thing one is testing and how one defines the sequence: in a let block because one may call it many times (using activate in each example) or in a before block because it is used once.  Whether the RSpec thread blocks or not is dependent on what you are testing.  In the LS case all input are blocking in some way.

I don't think a matcher can anticipate these scenarios.